core/storage/statistics/gae_models.py

```diff
@@ -1134,6 +1136,224 @@ def delete_playthroughs_multi(cls, playthrough_ids):
         cls.delete_multi(instances)
 
 
+class LearnerAnswerDetailsModel(base_models.BaseModel):
+    """Model for storing the answer details that a learner enters when they
+    are asked for it, per state one model instance is created. And currently
+    the model works for exploration and question.
+    """
+
+    # The maximum size in bytes that learner_answer_info_list can take,
+    # although the maximum size is 1000000 bytes, but 100000 bytes are
+    # left for other fields in the model.
+    _MAX_LEARNER_ANSWER_INFO_LIST_BYTE_SIZE = 900000
+
+    # The reference to the state for which model instance is created.
+    state_reference = ndb.StringProperty(required=True, indexed=True)
+    # The type of entity, that whether it is exploration, question, or any
+    # other entity if it is added in future.
+    entity_type = ndb.StringProperty(
+        required=True, indexed=True, choices=ENTITY_TYPE_CHOICES)
+    # The id of the interaction for which the answer details were received.
+    interaction_id = ndb.StringProperty(required=True, indexed=False)
+    # List of LearnerAnswerInfo dicts, which is defined in
+    # stats_domain.py, each dict corresponds to a single answer info of
+    # learner.
+    learner_answer_info_list = ndb.JsonProperty(repeated=True, indexed=False)
+    # The version of the learner_answer_info_list currently supported by Oppia.
+    # If the internal JSON structure of learner_answer_info_list changes,
+    # CURRENT_SCHEMA_VERSION in this class needs to be incremented.
+    schema_version = ndb.IntegerProperty(
+        indexed=True, default=(
+            feconf.CURRENT_LEARNER_ANSWERS_DETAILS_SCHEMA_VERSION))
+    # The total number of bytes needed to store all of the answers in the
+    # learner_answer_info_list. This value is found by summing the JSON
+    # sizes of all answer info dicts stored inside learner_answer_info_list.
+    accumulated_answer_info_json_size_bytes = ndb.IntegerProperty(
```

Comment:
It's fine for this to be indexed=True

Reply:
Please respond to **all** review comments

Reply:
Done